tools/sv-report

```diff
@@ -272,177 +217,430 @@ meta_tags, meta_urls = readConfig(args.meta_tags)
-    runner_data["user_time"] = 0
-    runner_data["system_time"] = 0
-    runner_data["ram_usage"] = 0
-    runner_data["passed_time"] = 0
-    runner_data["passed_size"] = 0
 
-    for t in glob(os.path.join(args.logs, runner_name, "**/*.log"),
-                  recursive=True):
-        t_id = t[len(args.logs) + 1:]
+with open(args.log_template, "r") as templ:
+    log_template = jinja2env.from_string(templ.read())
+
+
+def renderLogToHTMLFile(
+        out_dir: str, log_html_path: str, test_result: TestResult,
+        test_log: Dict[str, str], log_content: str):
+    files_map: Dict[str, str] = {}
+    log_html_dir = os.path.dirname(log_html_path)
+    for input_file in test_result.input_files:
+        # Absolute path:
+        input_file_html = os.path.join(out_dir, input_file + ".html")
+        # Path relative to rendered log:
+        input_file_html = os.path.relpath(input_file_html, log_html_dir)
+
+        files_map[input_file] = urllib.parse.quote(input_file_html)
+
+    log_content = convertPathsToRelativeLinks(
+        str(markupsafe.escape(log_content)), log_html_dir)
+
+    os.makedirs(os.path.dirname(log_html_path), exist_ok=True)
+    with open(log_html_path, 'w') as f:
+        f.write(
+            log_template.render(
+                input_files_map=files_map,
+                result=test_result,
+                log=test_log,
+                content=log_content))
+
+
+with open(args.code_template, "r") as templ:
+    src_template = jinja2env.from_string(templ.read())
+
+
+def renderInputFileToHTMLFile(
+        out_dir: str, html_path: str, input_file_path: str):
+    formatter = HtmlFormatter(
+        full=False, linenos=True, anchorlinenos=True, lineanchors='l')
+
+    os.makedirs(os.path.dirname(html_path), exist_ok=True)
+
+    raw_code = ""
+    try:
+        with open(input_file_path, 'rb') as f:
+            raw_code = f.read()
+    except IOError:
+        logger.warning(f"Error when opening file {input_file_path}")
+        try:
+            with open(html_path, 'w') as out:
+                out.write('Error when opening file ' + input_file_path)
+        except:
+            pass
+        return
+
+    code = highlight(raw_code, lex, formatter)
+    csspath = os.path.join(out_dir, "code.css")
+    csspath = os.path.relpath(csspath, os.path.dirname(html_path))
+    with open(html_path, 'w') as out:
+        out.write(
+            src_template.render(
+                csspath=csspath, filename=input_file_path, code=code))
+
+
+def renderTagResultsConfig(
+        js_path: str, tool: str, tag: str, test_results: Iterable[TestResult]):
+    tool = tool.lower()
+    tag = tag.lower()
+    js_global_variable_name = f"config_loader_data['{tool}/{tag}']"
+
+    # Order of values in each entry:
+    # [name, status, log_html_path, first_input_file_html_path]
+    config = []
+    for test_result in test_results:
+        name = test_result.name
+        status = 1 if test_result.status == TestStatus.PASSED else 0
```

Comment:
`status = int(test_result.status == TestStatus.PASSED)` if not, then I think it'd be better to expand this if

Reply:
Fixed.